### PR TITLE
Add Image Push Job For Kubernetes CSI 

### PR DESCRIPTION
**Pull request by vitt-bagal**
Adding image push job for the Kubernetes CSI packages for automating Image build, push(to staging area) process .

### CODE REVIEW

**Comment by vitt-bagal** [comment]
@pohly Could you please have a look ?
References: https://github.com/kubernetes-csi/node-driver-registrar/pull/67 and https://github.com/kubernetes-csi/node-driver-registrar/pull/68

File: config/jobs/image-pushing/k8s-staging-csi.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  kubernetes-csi/node-driver-registrar:
+    # The name should be changed to match the repo name above
+    - name: post-node-driver-registrar-push-images
+      cluster: test-infra-trusted
+      annotations:
+        # This is the name of some testgrid dashboard to report to.
+        testgrid-dashboards: sig-storage-csi-node-driver-registrar
+      decorate: true
+      branches:
+        - ^master$
```
Comment (pohly) [inline config/jobs/image-pushing/k8s-staging-csi.yaml]: Can you add a comment explaining that this is for publishing the "canary" images, something that we only do on the master branch?

File: config/jobs/image-pushing/k8s-staging-csi.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+postsubmits:
+  # This is the github repo we'll build from. This block needs to be repeated
+  # for each repo.
```
Comment (pohly) [inline config/jobs/image-pushing/k8s-staging-csi.yaml]: Eventually we'll probably want to have a generator script for all the various repos. There are too many to of them to create and maintain this file manually. But for now as a test it's okay.

File: config/jobs/image-pushing/k8s-staging-csi.yaml (hunk 1 above)
Comment (vitt-bagal) [reply config/jobs/image-pushing/k8s-staging-csi.yaml]: Added comment

**Comment by vitt-bagal** [comment]
@pohly @Katharine Looking at [prow](https://prow.k8s.io/?repo=kubernetes-csi%2Fnode-driver-registrar) it seems that postsubmit job for nodedriver registrar is not executed. Do we need to wait for a commit to be merged to node driver registrar master ?

**Comment by BenTheElder** [comment]
Yes. Postsubmits run post submit (after a merge), there's also a branch regex optionally that would restrict to only certain pushes.

**Comment by vitt-bagal** [comment]
Thanks @BenTheElder 

@pohly As Postsubmit job runs after a merge, Is it possible for us to commit something in node driver registrar master ? e.g. sync release tool ?

**Comment by pohly** [comment]
I created https://github.com/kubernetes-csi/node-driver-registrar/pull/71

**Comment by vitt-bagal** [comment]
@pohly  @BenTheElder  Image push job for node-driver-registrar got triggered after [PR#71](https://github.com/kubernetes-csi/node-driver-registrar/pull/71), however it fails: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-node-driver-registrar-push-images/1247569283886616576 Could you please check once?

**Comment by BenTheElder** [comment]
Hi, please check the "artifacts" link to see more logs. I unfortunately really don't have the bandwidth to debug everyone's CI.

**Comment by vitt-bagal** [comment]
The [build.log](https://storage.googleapis.com/kubernetes-jenkins/logs/post-node-driver-registrar-push-images/1247569283886616576/artifacts/build.log) from artifacts point to a url 'https://console.cloud.google.com/cloud-build/builds/ac677e83-ad64-4ec3-8204-f0727b714682?project=874328413592' which gives us error _**You do not have sufficient permissions to view this page**_. @pohly Can you access the logs?

**Comment by BenTheElder** [comment]
the CI robot has permissions and the tooling should be downloading all logs.
someone in wg-k8s-infra owns this project.

**Comment by spiffxp** [comment]
@pohly @vitt-bagal you should probably open a PR against https://github.com/kubernetes/k8s.io/blob/master/groups/groups.yaml#L705-L712, I'm not aware of the history of how that group was created, but I doubt it should be just @dims.  That would let you view the cloud-build link.

But I agree I'm confused why the logs weren't downloaded to `artifacts/` in the first place.  Maybe a flake.

Here's what I'm able to see
```
FETCHSOURCE
Fetching storage object: gs://k8s-staging-csi-gcb/source/1586278720.99-3d318855304f43258f3cef5b72318550.tgz#1586278721356113
Copying gs://k8s-staging-csi-gcb/source/1586278720.99-3d318855304f43258f3cef5b72318550.tgz#1586278721356113...
/ [0 files][    0.0 B/  1.7 MiB]                                                
/ [1 files][  1.7 MiB/  1.7 MiB]                                                
Operation completed over 1 objects/1.7 MiB.                                      
BUILD
Pulling image: gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4
v20190906-745fed4: Pulling from k8s-testimages/gcb-docker-gcloud
9d48c3bd43c5: Pulling fs layer
7f94eaf8af20: Pulling fs layer
9fe9984849c1: Pulling fs layer
cf0db633a67d: Pulling fs layer
0f7136d71739: Pulling fs layer
36b51914f992: Pulling fs layer
b5f72b4b9a52: Pulling fs layer
cf0db633a67d: Waiting
0f7136d71739: Waiting
36b51914f992: Waiting
b5f72b4b9a52: Waiting
9d48c3bd43c5: Verifying Checksum
9d48c3bd43c5: Download complete
9fe9984849c1: Verifying Checksum
9fe9984849c1: Download complete
7f94eaf8af20: Verifying Checksum
7f94eaf8af20: Download complete
0f7136d71739: Verifying Checksum
0f7136d71739: Download complete
36b51914f992: Verifying Checksum
36b51914f992: Download complete
9d48c3bd43c5: Pull complete
7f94eaf8af20: Pull complete
9fe9984849c1: Pull complete
cf0db633a67d: Verifying Checksum
cf0db633a67d: Download complete
b5f72b4b9a52: Verifying Checksum
b5f72b4b9a52: Download complete
cf0db633a67d: Pull complete
0f7136d71739: Pull complete
36b51914f992: Pull complete
b5f72b4b9a52: Pull complete
Digest: sha256:f17f1d2a58b8273e184a7c9f89a91d0193a6f458167891115868781610467ddd
Status: Downloaded newer image for gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4
gcr.io/k8s-testimages/gcb-docker-gcloud:v20190906-745fed4
unknown flag: --use
See 'docker --help'.
```

**Comment by dims** [comment]
This was way before we had automation:
https://github.com/kubernetes/k8s.io/pull/248/files#diff-8b242825ad9027b6efead6a412418c51R72-R77

It was mostly created as a placeholder to try things and no one got added afterwards. +1 to update the list of folks

**Comment by pohly** [comment]
> +1 to update the list of folks

So who should be on that list?

**Comment by vitt-bagal** [comment]
There were 2 builds triggered yesterday and for both, logs not downloaded by tooling again. Till the group membership is sorted, can @dims or @spiffxp  help us with the logs please?

**Comment by BenTheElder** [comment]
/uncc
I don't own these resources, leaving it to dims and spiffxp.

**Comment by dims** [comment]
the build is failing as we have a older docker version, see full log : https://paste.centos.org/view/raw/f725c290

i have a PR in test-infra for bumping gcb-docker-gcloud image to newer golang and docker - https://github.com/kubernetes/test-infra/pull/17299 

As for the list of people for that ACL for the job, please add all the leads for sig-storage, technical leads and any one that takes care of the csi images

**Pull request merged**